### DISCUSSION TITLE
Is there a way to restart a deployment controlled by inference service in non disruptive manner?

### DISCUSSION THREAD

**Discussion by endiczek** [kubeflow/kfserving#4104] [General]
Hello,

Checked documentation, googled, but cannot find a way to complete the task. 

## Pre isvc way 

From time to time there is a need to update worker nodes to newer version. To do it in non disruptive way with standard deployment my steps are as follows:
1. cordon node
2. rollout restart all deployments that have pods on the node
3. wait until the cordoned node has no pods left
4. update node
5. move to next node

## isvc way

With inference service there is no way to rollout restart a deployment if controlled by isvc.  Additionally it is not advised as described here: https://github.com/kserve/kserve/issues/2276#issuecomment-1166425939

I can edit isvc and increase minReplicas and maxReplicas by one. But this approach is cumbersome and not easy to automate.
Canary or other rollout mechanisms are not the choice as my model will be the same, so same name, same pvc, version, everything.

## Expected behaviour

Inference service has mechanisms to rollout new versions, like canary, A/B, that I believe are more demanding programatically than simpler rollout restart.
Is there a way to restart pods in deployment managed by isvc that will wait for new pods to be ready before terminating old pods?
And if there is no such a way is it possible to ask if described functionality can be added?

Please help.

**Comment by boarder7395** [comment]
I use kserve in rawDeployment mode but what rollout restart does under the hood is adds an annotation to the deployment. You can mimic this with kserve inference service triggering a clean update with:


kubectl -n {namespace} patch inferenceservice {inference-service-name} -p "{\"metadata\":{\"annotations\":{\"kubectl.kubernetes.io/restartedAt\":\"$(date -u +'%Y-%m-%dT%H:%M:%SZ')\"}}}" --type=merge

**Comment by spolti** [comment]
KServe defaults to rolling strategy that aims 0 downtime, for LLM, it might be problematic if there is enough GPUs available.
